airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -515,6 +515,7 @@ def load_string(
         encrypt: bool = False,
         encoding: Optional[str] = None,
         acl_policy: Optional[str] = None,
+        gzip: bool = False,
```

Comment:
Why only gzip? zstd is also gaining more and more popularity and other algorithms may be developed in the future. Have you considered approaching the interface in a general way but implement one algorithm for now?

Reply:
As it is mentioned in the issue  [AIRFLOW-3723](https://issues.apache.org/jira/browse/AIRFLOW-3723) to use gzip. As well as I noticed that most of the people in nearby code are using gzip so to maintain consistency I used gzip.

- https://github.com/apache/airflow/blob/master/airflow/providers/amazon/aws/hooks/s3.py#L495-L500
- https://github.com/apache/airflow/search?q=gzip

If you think changing the function parameter from `gzip` to `zip` would be better here in the same PR I will be happy to change it. Please let me know if you have more suggestions.

Reply:
I am not questioning the choice of the gzip algorithm. It is now standard and popular. I wonder if adding the "gzip" parameter instead of - for example - "compression" (enum with value "gzip") will not make this operator have many parameters in the future when more algorithm might be added. See for example at https://github.com/apache/airflow/blob/349b0811c3022605426ba57d30936240a7c2848a/airflow/providers/google/cloud/transfers/bigquery_to_gcs.py#L47-L48 regarding future-proof `compression` parameter.

I would like to point out that your experience with Airflow is definitely much greater than mine. I mainly observe and comment on the project from time to time.

Reply:
Thanks for the direction. 
Updated the pull request. Can you please take a look again?

Reply:
SGTM - using `compression` instead of `gzip`. 👍